crates/ruff_linter/src/noqa.rs

```diff
@@ -694,14 +695,17 @@ impl<'a> NoqaDirectives<'a> {
             }
         }
 
-        // Extend a mapping at the end of the file to also include the EOF token.
-        if let Some(last) = directives.last_mut() {
-            if last.range.end() == locator.contents().text_len() {
-                last.range = last.range.add_end(TextSize::from(1));
-            }
-        }
+        // Record whether last directive should include EOF token.
+        let last_directive_includes_eof = if let Some(last) = directives.last() {
+            last.range.end() == locator.contents().text_len()
+        } else {
+            false
+        };
 
-        Self { inner: directives }
+        Self {
+            inner: directives,
+            last_directive_includes_eof,
```

Comment:
Nit: I think I would rename this to `includes_end` to make it clear that the comparison should include the end range.